integration-tests/src/tests/client/resharding.rs

```diff
@@ -427,26 +427,35 @@ impl TestReshardingEnv {
                     }
                     *new_block.header().prev_hash()
                 }
-                None => self.env.clients[0].chain.head().unwrap().last_block_hash,
+                None => client.chain.head().unwrap().last_block_hash,
             };
-            // check that the target chunks in all prev blocks are not new
+
+            // Check that the target chunks are not new for all blocks between
+            // last_block_hash_with_empty_chunk (inclusive) and
+            // block_hash (exclusive).
             while &last_block_hash_with_empty_chunk != block_hash {
-                let last_block = self.env.clients[0]
-                    .chain
-                    .get_block(&last_block_hash_with_empty_chunk)
-                    .unwrap()
-                    .clone();
+                let last_block_hash = last_block_hash_with_empty_chunk;
+                let last_block = client.chain.get_block(&last_block_hash).unwrap().clone();
                 let chunks = last_block.chunks();
-                if chunks.len() == num_shards {
-                    assert_ne!(
-                        chunks.get(shard_id).unwrap().height_included(),
-                        last_block.header().height()
-                    );
+
+                let target_shard_ids = if chunks.len() == num_shards {
+                    // same shard layout between block and last_block
+                    vec![shard_id as ShardId]
                 } else {
-                    for chunk in chunks.iter() {
-                        assert_ne!(chunk.height_included(), last_block.header().height());
-                    }
```

Comment:
This check was overly ambitious for V2. When testing the V1 resharding, it's all or nothing when it comes to chunk production. In the V2 resharding, which happens to include `ChunkOnlyProducers`, chunks in a block can be produced by different chunk producers and so some can be present and some missing. 

I changed it to have the same condition for both cases, when the shard layout is the same and when it is different. The condition now is that the shards with ids `target_shard_ids` are missing in all blocks from `block_hash` to `last_block_hash_with_empty_chunk`.